src/utils/contexts/balances/hook.tsx

```diff
@@ -0,0 +1,44 @@
+import { CurrencyExt } from '@interlay/interbtc-api';
+import { AccountId } from '@polkadot/types/interfaces';
+import { useErrorHandler } from 'react-error-boundary';
+import { useQuery, UseQueryResult } from 'react-query';
+import { useSelector } from 'react-redux';
+
+import { StoreType } from '@/common/types/util.types';
+import { useGetCurrencies } from '@/utils/hooks/api/use-get-currencies';
+import useAccountId from '@/utils/hooks/use-account-id';
+
+import { BalanceData } from './types';
+
+const BALANCES_QUERY_KEY = 'getBalances';
+
+const getBalances = async (currencies: CurrencyExt[], accountId: AccountId): Promise<BalanceData> => {
+  const chainBalances = await Promise.all(
+    currencies.map((currency) => window.bridge.tokens.balance(currency, accountId))
+  );
+
+  return chainBalances.reduce(
+    (acc, balance) => ({
+      ...acc,
+      [balance.currency.ticker]: balance
+    }),
+    {}
+  );
+};
+
+const useGetBalances = (): UseQueryResult<BalanceData | undefined> => {
+  const { bridgeLoaded } = useSelector((state: StoreType) => state.general);
+  const accountId = useAccountId();
+  const { data: currencies, isSuccess } = useGetCurrencies(bridgeLoaded);
+  const queryResult = useQuery({
+    queryKey: BALANCES_QUERY_KEY,
```

Comment:
The query key should include account address too - it has to be unique to the query data, so the requests for different accounts are not having the same key.

Reply:
Even after having our subscriptions updating this key?

Reply:
When we change accounts, the subscription is already updating the query key

Reply:
Yes, I think it would be better to follow best practices of separating the keys for different requests (even if they update manually so it works). But if you have stronger opinion on keeping it this way I'm not against it :)